crates/flipperzero/src/gui/mod.rs

```diff
@@ -1,3 +1,230 @@
 //! GUI service.
 
 pub mod canvas;
+pub mod icon;
+pub mod icon_animation;
+pub mod view;
+pub mod view_port;
+pub mod xbm;
+
+use crate::{
+    gui::{
+        canvas::CanvasView,
+        view_port::{ViewPort, ViewPortCallbacks},
+    },
+    input::InputEvent,
+};
+use core::{
+    ffi::c_char,
+    fmt::Debug,
+    ops::{Deref, DerefMut},
+};
+use flipperzero_sys::{
+    self as sys, furi::UnsafeRecord, Canvas as SysCanvas, Gui as SysGui, GuiLayer as SysGuiLayer,
+};
+
+/// System Gui wrapper.
+pub struct Gui {
+    raw: UnsafeRecord<SysGui>,
+}
+
+impl Gui {
+    /// Furi record corresponding to GUI.
+    pub const RECORD: *const c_char = sys::c_string!("gui");
+
+    pub fn new() -> Self {
+        // SAFETY: `RECORD` is a constant
+        let gui = unsafe { UnsafeRecord::open(Self::RECORD) };
```

Comment:
It occurs to me that records are probably not supposed to be opened by multiple threads simultaneously. If that is the case, then separately from this PR, we might end up defining the SDK-provided records in some common location, where we can control access via a mutex. An alternative approach would be to create a harness for defining "record singletons", which would store them as heap statics (#64); then each service module would use it to define its record type.

The context for the above thought was "is it okay to call `Gui::new` twice?". Nothing in the current API will prevent it, but if the backing records are changed to be mutex-gated, then `Gui::new` would necessarily also be. And if we instead provide a record singleton harness, then `Gui::new` would become a private method.